ksql-parser/src/main/java/io/confluent/ksql/parser/AstBuilder.java

```diff
@@ -652,6 +654,10 @@ public Node visitSelectSingle(SqlBaseParser.SelectSingleContext context) {
     return new SingleColumn(getLocation(context), selectItemExpression, alias);
   }
 
+  private String replaceDotFieldRef(String input) {
+    return input.replace(DOT, "_").replace(STRUC_FIELD_REF, "__");
```

Comment:
Are we worried about this generating ambiguous names? One place this might trip us up is in aggregation.

Reply:
In the aggregates we prefix internal columns with `KSQL_AGG_VARIABLE_` so I think it should be easy to differentiate.